pkg/controller/releasemanager/plan/syncSLORules.go

```diff
@@ -0,0 +1,120 @@
+package plan
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/go-logr/logr"
+	picchuv1alpha1 "go.medium.engineering/picchu/pkg/apis/picchu/v1alpha1"
+	"go.medium.engineering/picchu/pkg/plan"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	rgNameRegex = regexp.MustCompile("[^a-zA-Z0-9]+")
+)
+
+type SyncSLORules struct {
+	App                         string
+	Namespace                   string
+	Labels                      map[string]string
+	ServiceLevelObjectiveLabels picchuv1alpha1.ServiceLevelObjectiveLabels
+	ServiceLevelObjectives      []*picchuv1alpha1.ServiceLevelObjective
+}
+
+func (p *SyncSLORules) Apply(ctx context.Context, cli client.Client, scalingFactor float64, log logr.Logger) error {
+	sloRules, err := p.SLORules()
+	if err != nil {
+		return err
+	}
+
+	if len(sloRules) > 0 {
+		for _, pr := range sloRules {
+			if err := plan.CreateOrUpdate(ctx, log, cli, &pr); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// SLORules returns a PrometheusRuleList of recording rules to support tracking of the ServiceLevelObjectives
+func (p *SyncSLORules) SLORules() ([]monitoringv1.PrometheusRule, error) {
+	prs := []monitoringv1.PrometheusRule{}
+
+	rule := p.prometheusRule()
+
+	for _, slo := range p.ServiceLevelObjectives {
```

Comment:
Same as above. Doesn't look like an issue, but might be safer to just stick with indexing to prevent future bugs?

Reply:
I switched these loops to all use the index.